app/main.py

```diff
@@ -1,6 +1,31 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    def wrapper(*args, **kwargs) -> Any:
+        flag = True
+        while flag:
+            data_iterable = tuple(
+                [item for item in args if isinstance(item, (int, str, float))]
+            )
+            if data_iterable not in cache_dict:
+                print("Calculating new result")
+                cache_dict[data_iterable] = func(*args, **kwargs)
+                return cache_dict[data_iterable]
+            else:
```

Comment:
`else` operator is redundant. remove it

Reply:
If I delete else how can I get the cached element back?